test/cucumber-tests/src/test/java/org/wso2/apk/integration/api/BaseSteps.java

```diff
@@ -60,6 +67,53 @@ public void theResponseStatusCodeShouldBe(int expectedStatusCode) {
         Assert.assertEquals(expectedStatusCode, actualStatusCode);
     }
 
+    @Then("I send {string} request to {string} with body {string}")
+    public void sendHttpRequest(String httpMethod, String url, String body) throws IOException {
+        if (sharedContext.getResponse() instanceof CloseableHttpResponse) {
+            ((CloseableHttpResponse) sharedContext.getResponse()).close();
+        }
+        if (CurlOption.HttpMethod.GET.toString().toLowerCase().equals(httpMethod.toLowerCase())) {
+            sharedContext.setResponse(httpClient.doGet(url, sharedContext.getHeaders()));
+        } else if (CurlOption.HttpMethod.POST.toString().toLowerCase().equals(httpMethod.toLowerCase())) {
+            sharedContext.setResponse(httpClient.doPost(url, sharedContext.getHeaders(), body, null));
+        } else if (CurlOption.HttpMethod.PUT.toString().toLowerCase().equals(httpMethod.toLowerCase())) {
+            sharedContext.setResponse(httpClient.doPut(url, sharedContext.getHeaders(), body, null));
+        } else if (CurlOption.HttpMethod.DELETE.toString().toLowerCase().equals(httpMethod.toLowerCase())) {
+            sharedContext.setResponse(httpClient.doPut(url, sharedContext.getHeaders(), body, null));
+        }
+    }
+
+    @Then("I set headers")
+    public void setHeaders(DataTable dataTable) {
+        List<List<String>> rows = dataTable.asLists(String.class);
+        for (List<String> columns : rows) {
+            String key = columns.get(0);
+            String value = columns.get(1);
+            key = Utils.resolveVariables(key, sharedContext.getValueStore());
+            value = Utils.resolveVariables(value, sharedContext.getValueStore());
+            sharedContext.addHeader(key, value);
+        }
+    }
+
+    @Then("I eventually receive {int} response code, not accepting")
+    public void eventualSuccess(int statusCode, DataTable dataTable) throws IOException, InterruptedException {
+        List<Integer> nonAcceptableCodes = dataTable.asList(Integer.class);
+        if (sharedContext.getResponse().getStatusLine().getStatusCode() == statusCode) {
+            Assert.assertTrue(true);
+        } else {
+            HttpResponse httpResponse = httpClient.executeLastRequestForEventualConsistentResponse(statusCode,
+                    nonAcceptableCodes);
+            sharedContext.setResponse(httpResponse);
+            Assert.assertEquals(httpResponse.getStatusLine().getStatusCode(), statusCode);
+        }
+    }
+
+    @Then("I receive {int} response code")
```

Comment:
Shall we use the already defined `@Then("the response status code should be {int}")` to avoid duplication

Reply:
Done